drivers/gpio/gpio_intel_apl.c

```diff
@@ -304,6 +284,87 @@ static int gpio_intel_apl_config(struct device *dev, int access_op,
 	return 0;
 }
 
+static int gpio_intel_apl_pin_interrupt_configure(struct device *dev,
+		unsigned int pin, enum gpio_int_mode mode,
+		enum gpio_int_trig trig)
+{
+	const struct gpio_intel_apl_config *cfg = dev->config->config_info;
+	struct gpio_intel_apl_data *data = dev->driver_data;
+	u32_t raw_pin, cfg0, cfg1;
+	u32_t reg, reg_en;
+
+	/* no double-edge triggering according to data sheet */
+	if (trig == GPIO_INT_TRIG_BOTH) {
+		return -ENOTSUP;
+	}
+
+	if (pin > cfg->num_pins) {
+		return -EINVAL;
+	}
+	pin = k_array_index_sanitize(pin, cfg->num_pins + 1);
+
+	raw_pin = cfg->pin_offset + pin;
+
+	if (!check_perm(dev, raw_pin)) {
+		return -EPERM;
```

Comment:
This error code is not documented in GPIO API. If we need it we should update the API. I'm not familiar with this architecture. What exactly is happening here? Could we use `EINVAL`?

Reply:
Some of the pins are controlled by, for example, the power management controller. These pins cannot be manipulated. I can use `EINVAL`. Originally `EPERM` is used because the datasheet talks about permissions.

Reply:
I see you've modified the code to use `EINVAL` but the `EPERM` use case also makes sense. The pin is valid, but the driver cannot use it since its permanently assigned to a different peripheral without a possibility to 'steal' it. 

If you prefer `EPERM` feel free to revert the change and I will prepare the PR to update GPIO API. I'll leave it to you to decide.

Reply:
Let's leave it as `EINVAL` for now. It makes error checking on applications easier, as the pin cannot be configured anyway.